base/strings/basic.jl

```diff
@@ -113,7 +113,7 @@ julia> isvalid(str, 2)
 false
 
 julia> str[2]
-ERROR: StringIndexError: invalid character index
+ERROR: StringIndexError("αβγdef", 2)
```

Comment:
This could use a custom `showerror` method with an explanation.

Reply:
^ @StefanKarpinski